jumpscale/servers/threebot/threebot.py

```diff
@@ -407,15 +399,18 @@ def add(self, path: str = None, giturl: str = None, **kwargs):
         # if package.name in self.packages:
         #     raise j.exceptions.Value(f"Package with name {package.name} already exists")
 
-        self.packages[package.name] = {"name": package.name, "path": package.path, "giturl": package.giturl}
-
         # execute package install method
         package.install(**kwargs)
 
         # install package if threebot is started
-        if self.threebot.started:
-            self.install(package)
-            self.threebot.nginx.reload()
+        try:
+            if self.threebot.started:
+                self.install(package)
+                self.threebot.nginx.reload()
+            self.packages[package.name] = {"name": package.name, "path": package.path, "giturl": package.giturl}
+        except Exception as e:
+            j.exceptions.Runtime(f"Error happend during adding {package.name}, {str(e)}")
```

Comment:
I think this will mask any thing that can go wrong, and will only give us a `Runtime` exception.

Can we change the order of adding the package to `self.packages` without try/except?

Reply:
done